lib/stats/stats-change-per-second.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2002-2013 One Identity
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef STATS_CHANGE_PER_SECOND_H
+#define STATS_CHANGE_PER_SECOND_H
+
+#include "stats/stats-aggregator.h"
+
+typedef struct
+{
+  StatsCounterItem *counter;
+  atomic_gssize average;
+  atomic_gssize sum;
+  atomic_gssize last_message_count;
+
+  gssize duration; /* if the duration equals -1, thats mean, it count since syslog start */
+  gchar *name;
+} CPSLogic;
```

Comment:
Currently `StatsAggregatorCPS` is responsible for 3 different EPS/CPS durations.
I think it's OK for now, but we should refactor this in a future PR:

- `CPSLogic` should actually be renamed to `StatsAggregatorCPS`, and should have the proper methods;
- What we currently call as `StatsAggregatedCPS` could be reimplemented as a general stats aggregator _group_, which could hold any number of aggregator instances that share the same input.